pages/index.js

```diff
@@ -1,198 +1,75 @@
-            }
-
-            code {
-                background: #fafafa;
-                border-radius: 5px;
-                padding: 0.75rem;
-                font-size: 1.1rem;
-                font-family: Menlo, Monaco, Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono,
-                    Courier New, monospace;
-            }
-
-            .grid {
-                display: flex;
-                align-items: center;
-                justify-content: center;
-                flex-wrap: wrap;
-
-                max-width: 800px;
-                margin-top: 3rem;
-            }
-
-            .card {
-                margin: 1rem;
-                flex-basis: 45%;
-                padding: 1.5rem;
-                text-align: left;
-                color: inherit;
-                text-decoration: none;
-                border: 1px solid #eaeaea;
-                border-radius: 10px;
-                transition: color 0.15s ease, border-color 0.15s ease;
-            }
-
-            .card:hover,
-            .card:focus,
-            .card:active {
-                color: #0070f3;
-                border-color: #0070f3;
-            }
-
-            .card h3 {
-                margin: 0 0 1rem 0;
-                font-size: 1.5rem;
-            }
-
-            .card p {
-                margin: 0;
-                font-size: 1.25rem;
-                line-height: 1.5;
-            }
-
-            @media (max-width: 600px) {
-                .grid {
-                    width: 100%;
-                    flex-direction: column;
-                }
-            }
-        `}</style>
-
-        <style jsx global>{`
-            html,
-            body {
-                padding: 0;
-                margin: 0;
-                font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans,
-                    Droid Sans, Helvetica Neue, sans-serif;
-            }
-
-            * {
-                box-sizing: border-box;
-            }
-        `}</style>
-    </div>
-);
-
-export default Home;
+import { useState } from 'react';
+import {
+    Heading,
+    Text,
+    ThemeProvider,
+    Flex,
+    FormErrorMessage,
+    FormControl,
+    FormLabel,
+    Button,
+    Input,
+} from '@chakra-ui/core';
+import { withTheme } from 'emotion-theming';
+
+const Home = () => {
+    const [formData, setFormData] = useState({
+        username: '',
+    });
+    const [formErrors, setFormErrors] = useState({});
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const { username } = formData;
+        if (!username) {
+            setFormErrors({ username: 'You have to enter an username to play' });
+        } else {
+            const { username: usernameError, ...errors } = formErrors;
+            setFormErrors(errors);
+        }
+    };
+    const handleOnChangeFactory = (field) => (e) => setFormData({ ...formData, [field]: e.target.value });
```

Comment:
No need to add factory, you can keep the name of the field.

```
const handleOnChange= ({ target: { name, value } }) => {
    setFormData({ ...formData, [name]: value });
};
```

Reply:
Your version looses the original arity though